### PR TITLE
test (frontend): add unit tests

### PR DESCRIPTION
**Pull request by giovanni-vaccarino**
### Description

This PR introduces the unit tests for the frontend project. It includes both tests for API-related modules and React components. Moreover a new workflow have been added to integrate the frontend tests in the CI.

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did

### CODE REVIEW

**Review by berviantoleo — APPROVED** [review]

File: frontend/src/tests/Sidebar.test.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    expect(baseProps.openConfirmDeleteChatPopup).toHaveBeenCalledWith("chat-1");
+  });
+
+  it("renders active chat styling when activeChatId matches", () => {
+    render(<Sidebar {...baseProps} chatList={exampleChats} activeChatId="chat-2" />);
+    expect(screen.getByText("Hello world")).toBeInTheDocument();
+  });
+
+  it("truncates chat name when first message text exceeds 25 characters", () => {
+    const longMessageChat: ChatSession = {
```
Comment (cnu1812) [inline frontend/src/tests/Sidebar.test.tsx:87]: Add Negative Test for Truncation kinda like this 
```
const shortMessageChat: ChatSession = {
  id: "chat-short",
  messages: [{ id: "msg-1", sender: "user", text: "Short message" }],
  createdAt: "2024-01-01T00:00:00Z",
  isLoading: false,
};

render(<Sidebar {...baseProps} chatList={[shortMessageChat]} />);
expect(screen.getByText("Short message")).toBeInTheDocument();
```

File: frontend/src/tests/api_chatbot.test.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    });
+  });
+
+  describe("createChatSession", () => {
+    it("creates a session and returns the session id", async () => {
+      (callChatbotApi as jest.Mock).mockResolvedValueOnce({
+        session_id: "abc123"
+      });
+
+      const result = await createChatSession();
```
Comment (cnu1812) [inline frontend/src/tests/api_chatbot.test.ts:40]: Add a negative test for createChatSession, Right now, this is only test a valid response. What if the API returns a malformed or missing session_id?

File: frontend/src/tests/api_chatbot.test.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+      expect(result).toEqual({
+        id: "mock-uuid",
+        sender: "jenkins-bot",
+        text: "Fallback error message"
+      });
+    });
+  });
+
+  describe("deleteChatSession", () => {
```
Comment (cnu1812) [inline frontend/src/tests/api_chatbot.test.ts:94]: Add Failure Test for deleteChatSession

File: frontend/src/tests/chatbotStorage.test.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      createdAt: '2024-01-02T00:00:00Z',
+      isLoading: false,
+    },
+  ];
+
+  beforeEach(() => {
+    sessionStorage.clear();
+    jest.clearAllMocks();
+  });
+
```
Comment (cnu1812) [inline frontend/src/tests/chatbotStorage.test.ts:24]: Add Test for Empty Array in Storage. i.e test for when the key exists and contains [] (i.e., a valid but empty session list). It's worth asserting that the utility doesn't treat it as an error.

File: frontend/src/tests/chatbotStorage.test.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      );
+    });
+
+    it('returns empty array when saved data is an empty string', () => {
+      sessionStorage.setItem('chatbot-sessions', '');
+      const result = loadChatbotSessions();
+      expect(result).toEqual([]);
+    });
+  });
+
```
Comment (cnu1812) [inline frontend/src/tests/chatbotStorage.test.ts:54]: Add a test for invalid object shape.

**Review by cnu1812 — CHANGES_REQUESTED** [review]

**Comment by krisstern** [comment]
There are currently also some merge conflicts waiting to be resolved as well

File: frontend/src/tests/api_chatbot.test.ts (hunk 2 above)
Comment (giovanni-vaccarino) [reply frontend/src/tests/api_chatbot.test.ts:40]: If the API returns a bad response, shouldn't the responsability be of `callChatbotApi.ts` ?

File: frontend/src/tests/api_chatbot.test.ts (hunk 3 above)
Comment (giovanni-vaccarino) [reply frontend/src/tests/api_chatbot.test.ts:94]: (same for createChat): If the API returns a bad response, shouldn't the responsability be of `callChatbotApi.ts` ?

File: frontend/src/tests/api_chatbot.test.ts (hunk 2 above)
Comment (berviantoleo) [reply frontend/src/tests/api_chatbot.test.ts:40]: @giovanni-vaccarino Actually, we should consider every known return by API. So, the UI will have appropriate feedback to the user when the API behaves incorrectly. We can't expect the API to always return successfully.

File: frontend/src/tests/api_chatbot.test.ts (hunk 3 above)
Comment (cnu1812) [reply frontend/src/tests/api_chatbot.test.ts:94]: same goes here

**Review by berviantoleo — APPROVED** [review]

**Review by krisstern — APPROVED** [review]
LGTM
